src/global/variables/clauses/IntegerLiteral.java

```diff
@@ -1,26 +1,108 @@
 package global.variables.clauses;
 
+import global.tools.InvalidClauseException;
+import global.utils.RegexUtil;
 import global.variables.Clause;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class IntegerLiteral extends Clause {
+    private Map<String, Boolean> MODES;
```

Comment:
Woah this feels like a lot. I was imagining more like:
```java
class IntergerLiteral {
    private String mode;
    public static final RANGE = 'range';
    ...
}
```

then:
```java
this.modes = IntegerLiteral.RANGE;  // in a constructor
```

The map way feels like a lot for a value that can only be one of the modes at a time

Reply:
Yeah my bad I think I tried to implement an approach like TypeScript, but this makes sense.